### PR TITLE
ctb: Naming and comments to clarify interfaces usage

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

The `OptimismMintableERC20` token used two interfaces in its `supportsInterface()` function. These were confusingly named after the functions that they contain (ie. `IL1Token`, `IRemoteToken`), rather than the contracts these interfaces represent. The names hav been changed (to `IOptimismMintableERC20` and `ILegacyMintableERC20`), and comments have been added to clarify the intent.

**Tests**

Existing tests have been kept.

### CODE REVIEW

**Review by tynes — APPROVED** [review]

File: packages/contracts-bedrock/contracts/universal/OptimismMintableERC20.sol [hunk 1]
```diff
@@ -12,7 +12,7 @@ import "./SupportedInterfaces.sol";
  *         Designed to be backwards compatible with the older StandardL2ERC20 token which was only
  *         meant for use on L2.
  */
-contract OptimismMintableERC20 is ERC20 {
+contract OptimismMintableERC20 is IOptimismMintableERC20, ERC20 {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/contracts/universal/OptimismMintableERC20.sol:15]: Shouldn't we add is `ILegacyMintableERC20` here too?

File: packages/contracts-bedrock/contracts/universal/SupportedInterfaces.sol [hunk 2]
```diff
@@ -4,18 +4,30 @@ pragma solidity ^0.8.0;
...
 }
 
-interface IL1Token {
+/**
+ * @custom:legacy
+ * @title ILegacyMintableERC20
+ * @notice This interface was available on the legacy L2StandardERC20 contract. It remains available
+ *         on the OptimismMintableERC20 contract for backwards compatibility.
+ */
+interface ILegacyMintableERC20 {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/contracts/universal/SupportedInterfaces.sol:27]: Hmmm why is `l2Bridge` not part of this interface?

File: packages/contracts-bedrock/contracts/universal/OptimismMintableERC20.sol (hunk 1 above)
Comment (maurelian) [reply packages/contracts-bedrock/contracts/universal/OptimismMintableERC20.sol:15]: Do new tokens need to support the legacy interface?

File: packages/contracts-bedrock/contracts/universal/SupportedInterfaces.sol (hunk 2 above)
Comment (maurelian) [reply packages/contracts-bedrock/contracts/universal/SupportedInterfaces.sol:27]: We got to this point just by maintaining the previous ERC165 interface derivation: 
https://github.com/ethereum-optimism/optimism/blob/aead527d57858498a9ebfb305bd344f8411698bf/packages/contracts-bedrock/contracts/universal/OptimismMintableERC20.sol#L75-L77

Looking back at the legacy token's actual `supportsInterface` implementation, it doesn't include the `l2Bridge` method either: 

https://github.com/ethereum-optimism/optimism/blob/0c20a11177b3b6d0be74f2c183c58bb198aeadca/packages/contracts/contracts/standards/L2StandardERC20.sol#L35

File: packages/contracts-bedrock/contracts/universal/OptimismMintableERC20.sol (hunk 1 above)
Comment (smartcontracts) [reply packages/contracts-bedrock/contracts/universal/OptimismMintableERC20.sol:15]: Yes, I believe so: https://github.com/ethereum-optimism/optimism/pull/3439/files/910e6bb14be3a59233c5e1cc198d3e59e8fd3bcf#diff-b5041e011abda1d1d726b7fd58224f6cdded883611096c7ad26bd35149020593R540

File: packages/contracts-bedrock/contracts/universal/SupportedInterfaces.sol (hunk 2 above)
Comment (smartcontracts) [reply packages/contracts-bedrock/contracts/universal/SupportedInterfaces.sol:27]: I vote for `remoteToken` to be included in the new interface and no changes to the old interface

Comment (maurelian) [reply packages/contracts-bedrock/contracts/universal/SupportedInterfaces.sol:29]: `remoteToken` was already included, but I've now also added `bridge()` to the new interface.

File: packages/contracts-bedrock/contracts/universal/OptimismMintableERC20.sol (hunk 1 above)
Comment (maurelian) [reply packages/contracts-bedrock/contracts/universal/OptimismMintableERC20.sol]: Done in this commit: [`f51a359` (#3439)](https://github.com/ethereum-optimism/optimism/pull/3439/commits/f51a3595efd5ea704a2463e4182a03fda35ba326)

**Review by smartcontracts — APPROVED** [review]

**Pull request merged**